src/Shared-Src/Native/loader.cpp

```diff
@@ -443,59 +445,90 @@ namespace shared
 
     HRESULT Loader::HandleJitCachedFunctionSearchStarted(FunctionID functionId, BOOL* pbUseCachedFunction)
     {
+        // Some of the logging in this method is a little too verbode even for debug.
+        // Turn it on here if you need it for investigations.
+        constexpr bool ExtraVerboseLogging = false;

```

Comment:
Generally we turn on Debug logging to investigate. Why having an extra one ?

Reply:
This one need to rebuild everything to activate it, you cannot activate it by env. var.... tha's not really useful in the long term

Reply:
Agree. Previously it was in the code and commented out and @tonyredondo  recommended that I uncomment it for the investigation I made (which was to see how often this gets invoked).
I am happy to leave it there commented like it was. I only made the compile time switch because it is now in 2 places (fast path and normal path) and this allows turning this on at once.

What is your preference, @gleocadie ?

Reply:
Oh you mean the Debug statement that was commented and you replace with the `if (debugEnabled)`.  A bit hard to follow :)

To be honest, do it your way, but I think that we should have a proper tooling/mechanism to investigate easily. Tomorrow, when running in the customer environment, I thing asking to re-run with specific flags to get detailed information is a better approach that sending a patched version.
Another ticket in our backlog :)

Reply:
@gleocadie There two separate points here. I think you are right on both, I just wanna keep them separate. :)

a) The `Debug(..)` statement was commented out before the change (think, if was compile-time-switched). How that there are two _related_ Debug statements in this method I replaced the compile-time-switch based on comments with another compile-time-switch based on a method-scoped `constexpr`.

b) The `Debug(..)` statements in the entire class were taking cycles even if debug logging was disabled, because or the parameter formatting issue we discussed earlier. I addressed it with `logDebugIsEnabled` flag (earlier PR).

Point (b) definitely needs some improvement./refactoring in the mid-term, as discussed.

Point (a) is minor and local, I do not mind what way we go there. I think it's not worth putting much effort into. Here, I just explained why I did what I did, but I'd be happy to do it in any other way. :)